File: frontend/smart_delivery/lib/core/constants/api_urls.dart
class ApiUrls {
  // AUTH
  static const String login = '/auth/login/';
  static const String refresh = '/auth/refresh/';
  static const String register = '/auth/register/';

  // USERS
  static const String users = '/users/';

  // DELIVERY REQUESTS (client)
  static const String deliveryRequests = '/delivery-requests/';
  static const String availableRequests = '/delivery-requests/available/';

  // DELIVERIES (courier)
  static const String deliveries = '/deliveries/';
  static const String acceptDelivery = '/deliveries/accept/';
  static String updateDeliveryStatus(int id) => '/deliveries/$id/';
}

File: backend/config/api_urls.py
# config/api_urls.py
from django.urls import path, include
from rest_framework.routers import DefaultRouter

from delivery_requests.views import DeliveryRequestViewSet
from delivery.views import DeliveryViewSet

router = DefaultRouter()
router.register(r"delivery-requests", DeliveryRequestViewSet, basename="delivery-request")
router.register(r"deliveries", DeliveryViewSet, basename="delivery")

urlpatterns = [
    # API routers
    path("", include(router.urls)),

    # Accounts/auth endpoints
    path("", include("accounts.urls")),
]
